packages/api-codec/src/StorageKey.ts

```diff
@@ -2,8 +2,67 @@
 // This software may be modified and distributed under the terms
 // of the ISC license. See the LICENSE file for details.
 
+import { AnyU8a } from './types';
+
+import isFunction from '@polkadot/util/is/function';
+
 import Bytes from './Bytes';
+import { StorageFunctionMetadata } from './Metadata';
+
+export interface StorageFunction extends StorageFunctionMetadata {
+  create (arg?: any): StorageKey;
```

Comment:
Ok, check this - it is moved in from storage (and storage imports this), however... instead of `create (arg?: any): StorageKey` it is now, well, that.

The issue is in type-storage in the create function, I needed to create a new instance since the Object.assign we had for docs/type didn't actually work as expected. The stuff just didn't get pulled through. So now create it as a clone, then cast and assign the create.

I know it is uglier... However, we probably don't expect people to call this themselves...

Any suggestions welcome (when looking at type-storage/createFunction)

Reply:
Alternative would be something like -

```
export interface StorageFunction {
  (arg?: any): StorageKey;
  meta: StorageFunctionMetadata
}
```

There are benefits, i.e. can actually skip the new Metadata we do in createFunction. Also keeps the cleanliness of using the code should creation be needed.